### PR TITLE
[PLATFORM-1243] Close modals on route change

### PR DESCRIPTION
**Pull request by juhah**
Fixes issues where modals are left open after navigating away from the page, namely:
- Creating a new product and then hitting browser back button
- Clicking on purchase modal and then hitting browser back button

### CODE REVIEW

File: app/src/shared/contexts/ModalApi/index.jsx [hunk 1]
```diff
@@ -53,6 +54,20 @@ function useModalContext(): ContextProps {
         })
     }, [modals, isMounted])
 
+    // close all modals on route change
+    useEffect(() => () => {
+        if (!isMounted()) { return }
+
+        setModals((prevModals) => {
+            Object.keys(prevModals).forEach((modalId) => {
+                if (prevModals[modalId].reject && typeof prevModals[modalId].reject === 'function') {
```
Comment (mondoreale) [inline app/src/shared/contexts/ModalApi/index.jsx]: `typeof prevModals[modalId].reject === 'function'` should be enough.

**Comment by mondoreale** [comment]
Also gave it a try in browser. Works well!

@juhah, it's been open for 8 days. Can we merge soon?

File: app/src/shared/contexts/ModalApi/index.jsx (hunk 1 above)
Comment (juhah) [reply app/src/shared/contexts/ModalApi/index.jsx]: Left this as is finally, doesn't do any harm and the other conditions are similar.

**Comment by mondoreale** [comment]
🚀

**Comment by mondoreale** [comment]
@juhah, u wanna keep that branch? 🤔

**Pull request merged**